File: src/AcceptanceSpecSystem.Data/Entities/PromptTemplate.cs
namespace AcceptanceSpecSystem.Data.Entities;

public enum PromptTemplateScene
{
    Unknown = 0,
    MatchingReview = 1,
    ImportDuplicateReview = 2,
    MatchingGenerate = 3
}

/// <summary>
/// Prompt模板实体
/// </summary>
public class PromptTemplate
{
    /// <summary>
    /// 模板ID
    /// </summary>
    public int Id { get; set; }

    /// <summary>
    /// 模板名称（唯一）
    /// </summary>
    public string Name { get; set; } = "default";

    /// <summary>
    /// Prompt模板内容
    /// </summary>
    public string Content { get; set; } = string.Empty;

    /// <summary>
    /// 模板场景
    /// </summary>
    public PromptTemplateScene Scene { get; set; }

    /// <summary>
    /// 展示名称
    /// </summary>
    public string DisplayName { get; set; } = string.Empty;

    /// <summary>
    /// 是否为系统模板
    /// </summary>
    public bool IsSystem { get; set; }

    /// <summary>
    /// 是否为默认模板
    /// </summary>
    public bool IsDefault { get; set; }

    /// <summary>
    /// 创建时间
    /// </summary>
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

    /// <summary>
    /// 更新时间
    /// </summary>
    public DateTime? UpdatedAt { get; set; }
}

File: web/src/api/prompt-template.ts
import { http } from "@/utils/http";
import type { ApiResponse, PagedData, PagedRequest } from "./customer";

export interface PromptTemplate {
  id: number;
  name: string;
  scene: string;
  displayName: string;
  content: string;
  isSystem: boolean;
  isDefault: boolean;
  usageDescription: string;
  availableVariables: string[];
  createdAt: string;
  updatedAt?: string | null;
}

export interface UpdatePromptTemplateRequest {
  displayName: string;
  content: string;
}

export interface PromptTemplatePreviewRequest {
  scene: string;
  content: string;
}

export interface PromptTemplatePreviewResponse {
  isValid: boolean;
  errors: string[];
  renderedPrompt: string;
  exampleJson?: string | null;
  structuredOutputIsValid: boolean;
  structuredOutputError?: string | null;
}

export interface PromptTemplateListRequest extends PagedRequest {}

const baseUrl = "/api/prompt-templates";

export const getPromptTemplateList = (params?: PromptTemplateListRequest) => {
  return http.request<ApiResponse<PagedData<PromptTemplate>>>("get", baseUrl, {
    params
  });
};

export const updatePromptTemplate = (
  id: number,
  data: UpdatePromptTemplateRequest
) => {
  return http.request<ApiResponse<PromptTemplate>>("put", `${baseUrl}/${id}`, {
    data
  });
};

export const previewPromptTemplate = (
  data: PromptTemplatePreviewRequest
) => {
  return http.request<ApiResponse<PromptTemplatePreviewResponse>>(
    "post",
    `${baseUrl}/preview`,
    { data }
  );
};

export const resetSystemPromptTemplate = (scene: string) => {
  return http.request<ApiResponse<PromptTemplate>>(
    "post",
    `${baseUrl}/reset-system/${scene}`
  );
};
